zerver/lib/bugdown/__init__.py

```diff
@@ -1785,6 +1785,14 @@ def get_full_name_info(realm_id: int, full_names: Set[str]) -> Dict[str, FullNam
     if not full_names:
         return dict()
 
+    # Remove the trailing part of the `user|id` mention syntax.
+    name_re = r'(.+)\|\d+$'
```

Comment:
Use named groups (the `(?P<user_id>)` syntax); we use those in all of our backend regular expressions for better readability.